libensemble/executors/executor.py

```diff
@@ -254,6 +254,25 @@ def wait(self, timeout=None):
 
         self._set_complete()
 
+    def result(self, timeout=None):
+        """Wrapper for task.wait() that also returns the task's status on completion.
+
+        Parameters
+        ----------
+
+        timeout:
+            Time in seconds after which a TimeoutExpired exception is raised"""
+        self.wait(timeout=timeout)
+        return self.state
+
+    def running(self):
+        """ Return ```True`` if task is currently running."""
+        return self.state == 'RUNNING'
+
+    def done(self):
+        """ Return ```True`` if task is finished or successfully cancelled."""
+        return self.state in ['FINISHED', 'USER_KILLED']
```

Comment:
I'm wondering if our state of FAILED should also return True? 
I may have made this a bit confusing. I think self.finished == True includes FAILED, but a self.state of FINISHED is different to a state of FAILED. I think wrapping self.finished may be what we want.